test/conftest.py

```diff
@@ -0,0 +1,60 @@
+#   FirmwareUploader
+#   Copyright (c) 2021 Arduino LLC.  All right reserved.
+
+#   This library is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Lesser General Public
+#   License as published by the Free Software Foundation; either
+#   version 2.1 of the License, or (at your option) any later version.
+
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+import pathlib
+import platform
+import invoke.context
+import pytest
+
+
+@pytest.fixture(scope="function")
+def run_command(pytestconfig, working_dir):
+    """Provide a wrapper around invoke's `run` API so that every test will work in the same temporary folder.
+    Useful reference:
+        http://docs.pyinvoke.org/en/1.4/api/runners.html#invoke.runners.Result
+    """
+
+    fwuploader_path = pathlib.Path(pytestconfig.rootdir).parent / "FirmwareUploader"
+
+    def _run(cmd_string, custom_working_dir=None, custom_env=None):
+        if not custom_working_dir:
+            custom_working_dir = working_dir
+        cli_full_line = '"{}" {}'.format(fwuploader_path, cmd_string)
+        run_context = invoke.context.Context()
+        # It might happen that we need to change directories between drives on Windows,
+        # in that case the "/d" flag must be used otherwise directory wouldn't change
+        cd_command = "cd"
+        if platform.system() == "Windows":
+            cd_command += " /d"
+        # Context.cd() is not used since it doesn't work correctly on Windows.
+        # It escapes spaces in the path using "\ " but it doesn't always work,
+        # wrapping the path in quotation marks is the safest approach
+        with run_context.prefix(f'{cd_command} "{custom_working_dir}"'):
+            return run_context.run(
+                command=cli_full_line, echo=False, hide=True, warn=True, env=custom_env, encoding="utf-8"
+            )
+
+    return _run
```

Comment:
I'd prefer if we structure the `run_command` similarly to the `arduino-lint` one, this one is more similar to the one used by the Arduino CLI and it might fail on Windows cause of spaces in paths.

https://github.com/arduino/arduino-lint/blob/85638b82451864b1dbe35774c7da1bc5eea64e6e/test/test_all.py#L261-L295

Reply:
:facepalm: I took the one from [there](https://github.com/arduino/arduino-lint/pull/101). I should have checked if it was updated later. Thanks, I'll apply the suggestion